app/uk/gov/hmrc/merchandiseinbaggage/service/DeclarationService.scala

```diff
@@ -49,18 +49,34 @@ class DeclarationService @Inject()(
       }
   }
 
+  def amendDeclaration(declaration: Declaration)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Declaration] = {
+    val updatedDeclaration = updatePaymentStatusIfNeeded(declaration)
+    declarationRepository
+      .upsertDeclaration(updatedDeclaration)
+      .andThen {
+        case Success(result) if canTriggerEmailsAndAudit(result) =>
+          triggerEmailsAndAudit(result)
+      }
+  }
+
   private def canTriggerEmailsAndAudit(declaration: Declaration): Boolean =
     declaration.declarationType == Export || importWithNoPayment(declaration)
 
   private def updatePaymentStatusIfNeeded(declaration: Declaration) =
-    declaration match {
-      case d if importWithNoPayment(d) =>
-        d.copy(paymentStatus = Some(NotRequired))
+    declaration.declarationType match {
+      case Import if newDeclarationWithNoTaxDue(declaration) || amendDeclarationWithNoTaxDue(declaration) =>
+        declaration.copy(paymentStatus = Some(NotRequired))
```

Comment:
In the case of an amendDeclarationWithNoTaxDue you will want to set the actual Amendments paymentStatus to NotRequired, rather than the parent Declaration.paymentStatus

Reply:
oh yes, i thought i covered all, but missed this one :) , will fix it now